File: python/day01/day01.py
import os
import time


def main():
    # input
    print(os.getcwd())
    day = "01"
    year = "2021"
    inputFile = f'../inputs/day{day}.txt'
    print(inputFile)
    part1, part2 = 0, 0
    with open(inputFile) as f:
        lines = f.read().splitlines()

    input = list(map(int, lines))

    start_time = time.time()

    # part1 & 2
    part1 = sum(b > a for (a, b) in zip(input, input[1:]))
    part2 = sum(b > a for (a, b) in zip(input, input[3:]))

    duration = int((time.time() - start_time) * 1000000)

    print(
        f"AoC {year} - Day {day}:\n\nPart 1:\t{part1}\nPart 2:\t{part2}\nDuration:\t{duration} ns")


if __name__ == "__main__":
    main()

File: cpp/day01/day01.cpp
/*
Advent of Code 2021 - Day 01
[https://adventofcode.com/2021/day/1]

*/
#include <bits/stdc++.h>
#include "reader.hpp"

using namespace std;

std::vector<std::string> split(const std::string &str, const char &delim = ',')
{
    std::stringstream ss(str);
    std::string token;
    std::vector<std::string> toks;
    while (std::getline(ss, token, delim))
    {
        toks.push_back(token);
    }
    return toks;
}

int main()
{
    
    // Part 1 - parse input to ints and return total value

    string input_file = "../../inputs/day01.txt";
    ifstream ifs(input_file, ifstream::in);
    auto input_lines = vector<string>(read_input(ifs));
    vector<int> input(2, INT_MAX);
    transform(input_lines.begin(), input_lines.end(), back_inserter(input),
              [](const string &s)
              { return stoi(s); });

    
    // Part1 & Part2
    auto it = input.begin() + 3;
    int part1(0), part2(0);
    for (; it != input.end(); it++)
    {
        part2 += (*it > *(it - 3) ? 1 : 0);
        part1 += (*it > *(it - 1) ? 1 : 0);
    }
    cout << "Advent of Code 2021 - Day 01\n\n";
    cout << "Part 1:\t"<< part1 << "\n" << "Part 2:\t"<<part2 << endl;

}